CHE-9244 Add VsCode vim extension to the plugin registry

vinokurig opened the pull request:
### What does this PR do?
Add [VsCode vim extension](https://marketplace.visualstudio.com/items?itemName=vscodevim.vim) to the plugin registry

https://github.com/eclipse/che/issues/9244

benoitf commented:
would be nice to have only lowercase and use already ms-code folder
`v3/plugins/vsCodeVim` --> `v3/plugins/ms-vscode/vim`

vinokurig commented:
done

benoitf commented:
@vinokurig sorry I edited my comment, it's not a microsoft one

benoitf commented on v3/plugins/ms-vscode/vim/v1.12.0/meta.yaml:
```diff
@@ -0,0 +1,15 @@
+publisher: VSCodeVim
```
publisher id and folder should be the same

else you've an error on build like

!!! Location mismatch in plugin 'VSCodeVim/vim/latest':
!!!   Expected location: 'v3/plugins/VSCodeVim/vim/latest/meta.yaml'
!!!   Actual location: 'v3/plugins/ms-vscode/vim/latest/meta.yaml'

vinokurig commented:
Yes, the publisher is VsCodeVim, so what folder name should we choose?

benoitf commented:
let's use like on VS Code marketplace: https://github.com/VSCodeVim/Vim/blob/master/package.json#L7

and then use `vscodevim` as folder name/publisher

vinokurig commented:
done

amisevsk commented on v3/plugins/vscodevim/vim/v1.12.0/meta.yaml:
```diff
@@ -0,0 +1,15 @@
+publisher: vscodevim
+apiVersion: v2
+name: vim
+version: v1.12.0
```
Minor nitpick/optional request: the other plugins generally avoid using `v` in version numbers. I'd prefer to stick to a `major.minor.bugfix` format where possible (in case we need to process version numbers at some later point).

vinokurig replied:
fixed

amisevsk commented:
Digging in a little deeper, I tried running on nightly, adding the plugin to the default java-maven devfile, and I see the error log
```
Extension-Host:Error on activation of vim Error: Unable to find .vimrc file
    at VimrcImpl.<anonymous> (/tmp/vscode-unpacked/vscodevim.vim.v1.12.0.wdhjahwzof.vim-1.12.0.vsix/extension/out/src/configuration/vimrc.js:27:23)
    at Generator.next (<anonymous>)
    at /tmp/vscode-unpacked/vscodevim.vim.v1.12.0.wdhjahwzof.vim-1.12.0.vsix/extension/out/src/configuration/vimrc.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/tmp/vscode-unpacked/vscodevim.vim.v1.12.0.wdhjahwzof.vim-1.12.0.vsix/extension/out/src/configuration/vimrc.js:4:12)
    at VimrcImpl.load (/tmp/vscode-unpacked/vscodevim.vim.v1.12.0.wdhjahwzof.vim-1.12.0.vsix/extension/out/src/configuration/vimrc.js:21:16)
    at Configuration.<anonymous> (/tmp/vscode-unpacked/vscodevim.vim.v1.12.0.wdhjahwzof.vim-1.12.0.vsix/extension/out/src/configuration/configuration.js:172:37)
    at Generator.next (<anonymous>)
    at /tmp/vscode-unpacked/vscodevim.vim.v1.12.0.wdhjahwzof.vim-1.12.0.vsix/extension/out/src/configuration/configuration.js:14:71
    at new Promise (<anonymous>)
```

benoitf commented:
please note that lot of bugs have only be fixed in theia upstream this morning
https://github.com/eclipse-theia/theia/pull/6687

vinokurig commented:
@amisevsk @benoitf 
I've updated the plugin source to latest and the error is gone. The latest version witch is available in the vscode marketplace is `1.12.2`, but the latest version in the Vim's GitHub repository release page is `1.12.0`. In order to have a download link for the latest `1.12.2` version I've forked the `VsCodeVim/Vim` repository to the `che-incubator` and added the latest `1.12.2`release version.

vinokurig commented:
@amisevsk 
> Typing with the plugin enabled is visibly slow -- it's not hard to type faster than letters appear on screen.

https://github.com/eclipse-theia/theia/pull/6687#pullrequestreview-329677352
> It doesn't seem to be possible to fully disable the plugin; even after using the `Vim: Toggle Vim mode` command, the plugin is still intercepting certain keys (e.g. `g`).

Fixed that in https://github.com/eclipse-theia/theia/pull/6752

amisevsk commented:
@vinokurig thanks, ping me once changes are pushed, in case I forget to re-review.

vinokurig commented:
@amisevsk https://github.com/eclipse-theia/theia/pull/6752 has been merged to Theia's master, but am not sure that they are present in the latest `che-theia` docker image at the moment

vinokurig commented:
@amisevsk @benoitf downgraded the plugin to 1.11.3 because of the bug in the latest version: https://github.com/VSCodeVim/Vim/issues/4396
